### PR TITLE
feat: add calculator functionality

### PR DESCRIPTION
**Pull request by Amrhub**
- [x] made `logic` directory and added `calculate.js` and `operate.js` for calculator logic
- [x] implemented event handlers in `KeypadRow` class then passed event target to parent `calculator.js`
- [x] tested it in the browser to make sure functionality works fine

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,24 +1,50 @@
...
         <div className="calculator__keypad">
-          <KeypadRow values={['AC', '+/-', '%', '÷']} />
-          <KeypadRow values={['7', '8', '9', 'x']} />
-          <KeypadRow values={['4', '5', '6', '-']} />
-          <KeypadRow values={['1', '2', '3', '+']} />
-          <KeypadRow values={['0', '.', '=']} />
+          <KeypadRow
+            values={['AC', '+/-', '%', '÷']}
+            callCalculate={this.callCalculate}
+          />
```
Comment (safafa) [inline src/components/Calculator.js:31]: - [x] When I try any number modulo zero the app crashes. Kindly fix the issue, maybe try displaying an error message like when the user tries division by zero.

**Review by safafa — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUESTED ♻️ 

Hi @Amrhub , You did a great job so far 👏🏼 👏🏼 

However, here are some points that you need to work on to get your project approved.

## Required changes :

- Linter checks are passing.
- Use of lifecycle methods.

- [ ] Check the inline comments.

## Optional changes :
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/Calculator.js (hunk 1 above)
Comment (Amrhub) [reply src/components/Calculator.js:31]: Thanks, That's a really good point

**Review by safafa — APPROVED** [review]
## STATUS: APPROVED 🟢 👏
Hello, @Amrhub  great job working on the required changes 👍 💯

The app is working as expected. Now you can merge this branch 💪 😄
HAPPY CODING!! AND GOOD LUCK WITH THE NEXT ONES 🤞 🍀

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**